src/main/python/janusgraph_python/structure/io/GraphsonReader.py

```diff
@@ -0,0 +1,91 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from gremlin_python.structure.io.graphsonV3d0 import GraphSONReader
+from gremlin_python.structure.io.graphsonV3d0 import GraphSONUtil
+from ...serializer.RelationIdentifierDeserializer import RelationIdentifierDeserializer
+
+
+class JanusGraphSONReader(object):
+    """
+    This class registers JanusGraph-specific deserializers so that objects like GeoShape, RelationIdentifier
+    can be interpreted on Python client side.
+    """
+
+    GRAPHSON_PREFIX = "janusgraph"
+    GEO_GRAPHSON_BASE_TYPE = "Geoshape"
+    RELATIONID_BASE_TYPE = "RelationIdentifier"
+    GeoShape_GRAPHSON_TYPE = GraphSONUtil.formatType(GRAPHSON_PREFIX, GEO_GRAPHSON_BASE_TYPE)
+    RelationID_GRAPHSON_TYPE = GraphSONUtil.formatType(GRAPHSON_PREFIX, RELATIONID_BASE_TYPE)
+
+    deserializers = dict()
+
+    def __init__(self):
+        self.reader = None
+
+    def __register_default_deserializers(self):
+        """
+            This method is used to register the Default deserializers for JanusGraph's python client.
+            Currently the deserializer registers GeoShape and RelationIdentifier classes.
+
+        Returns:
+            None
+        """
+        janusDeSerializers = self.__build_deserializers()
+
+        self.deserializers.update(janusDeSerializers)
+
+    def __build_deserializers(self):
+        """ The actual method which takes care of adding JanusGraph specific deserializers.
+        Returns:
+            dict
+        """
+        # Currently the default de-serializers registered.
+
+        janusDeSerializers = {
+            self.RelationID_GRAPHSON_TYPE: RelationIdentifierDeserializer
+        }
+
+        return janusDeSerializers
+
+    def build(self):
+        """ The method registers JanusGraph specific deserializers into Gremlin GraphSON Reader class.
+
+        Returns:
+            GraphSONReader
+        """
+        self.__register_default_deserializers()
+        self.reader = GraphSONReader(self.deserializers)
+        return self.reader
+
+    def register_deserializer(self, typeClass, deserializer):
+        """ This method is used to registering any additional JanusGraph de-serializers.
+
+        Args:
+            typeClass (str): The identifier to be used with underlying graph to register the De-serializer against.
+            deserializer: The Deserializer class.
+
+        Returns:
+
+        """
+
+        objectIdentifier = GraphSONUtil.formatType(self.GRAPHSON_PREFIX, typeClass)
```

Comment:
It's wrong to use the `janusgraph` namespace here as the point of this method is to allow users to register a deserializer for their own types and they should use their own namespace.
So, just use the GraphSON type ID that the user gives you directly.

Reply:
Should I add GraphSON Prefix as parameter you say?
Like
```
    def register_deserializer(self, typeID, prefix, deserializer):
        """ This method is used to registering any additional JanusGraph de-serializers.

        Args:
            typeID (str): The identifier to be used to register the De-serializer against.
            prefix (str): The Namespace to use in typeID
            deserializer: The De-serializer class.

        Returns:

        """

        object_identifier = GraphSONUtil.formatType(prefix, typeID)

        self.deserializers[object_identifier] = deserializer

        return self
```

Reply:
I would just let the user provide a GraphSON type string as that's also how gremlin-python (and the other GLVs) does it. This GraphSON type should usually consist of the namespace and the type name, but users could also use a different format if they really want to.

Reply:
I see. So @FlorianHockmann instead you want to have 2 inputs here. `type string` and `deserializer class` not 3? 
We shall leave it to user to do `customNamespace:typeID` or `janusgraph:MOCK` or `GraphSONUtil.formatType(prefix, typeID)` outside of code is it?

Reply:
> We shall leave it to user to do customNamespace:typeID or janusgraph:MOCK or GraphSONUtil.formatType(prefix, typeID) outside of code is it?

Yes, users don't have to use a namespace at all if they don't want to. So I would just let them specify a `type_id` and they can use a namespace for that if they want, but they don't have to.